googleapiclient/discovery.py

```diff
@@ -355,8 +355,9 @@ def build_from_document(
     scopes = list(
       service.get('auth', {}).get('oauth2', {}).get('scopes', {}).keys())
 
-    # If so, then the we need to setup authentication.
-    if scopes:
+    # If so, then the we need to setup authentication if no developerKey is
+    # specified.
+    if scopes and not developerKey:
```

Comment:
If a `developerKey` is specified, does this make the value of `scopes` no longer relevant?
If so, is there a case for raising an exception if both are specified (and scopes is non-default)?